jobbergate-agent/jobbergate_agent/clients/influx.py

```diff
@@ -0,0 +1,27 @@
+"""Core module for defining the InfluxDB client."""
+
+from influxdb import InfluxDBClient
+from loguru import logger
+
+from jobbergate_agent.settings import SETTINGS
+
+
+def initialize_influx_client() -> None | InfluxDBClient:
+    """Initialize the InfluxDB client."""
+    if SETTINGS.INFLUX_INTEGRATION_ENABLED:
+        logger.debug("InfluxDB integration is enabled. Initializing InfluxDB client...")
+        return InfluxDBClient.from_dsn(
+            str(SETTINGS.INFLUX_DSN),
+            pool_size=SETTINGS.INFLUX_POOL_SIZE,
+            ssl=SETTINGS.INFLUX_SSL,
+            verify_ssl=SETTINGS.INFLUX_VERIFY_SSL,
+            timeout=SETTINGS.INFLUX_TIMEOUT,
+            udp_port=SETTINGS.INFLUX_UDP_PORT,
+            cert=SETTINGS.INFLUX_CERT_PATH,
+        )
+    else:
+        logger.debug("InfluxDB integration is disabled")
+        return None
+
+
+INFLUXDB_CLIENT = initialize_influx_client()
```

Comment:
I think that initializing every cycle is redundant because the integration won't be enabled from one cycle to the other. I agree that is weird manipulating a variable like that but at the same time I think this redundant behaviour is not ideal as well. Once the integration is set (True or False), it cannot be changed unless the entire agent stops (which I think must be expected).